include/boost/http_proto/serializer.hpp

```diff
@@ -297,44 +325,130 @@ class BOOST_SYMBOL_VISIBLE
 
 //------------------------------------------------
 
+/**
+    The type used for caller-provided body data during
+    serialization.
+*/
 struct serializer::stream
 {
     /** Constructor.
+
+        The only valid operations on default constructed
+        streams are assignment and destruction.
     */
     stream() = default;
 
     /** Constructor.
+
+        The constructed stream will share the same
+        serializer as `other`.
     */
-    stream(stream const&) = default;
+    stream(stream const& other) = default;
 
-    /** Constructor.
+    /** Assignment.
+
+        The current stream will share the same serializer
+        as `other`.
     */
-    stream& operator=
-        (stream const&) = default;
+    stream& operator= (
+        stream const& other) = default;
 
+    /**
+        A MutableBufferSequence consisting of a buffer pair.
+     */
     using buffers_type =
         buffers::mutable_buffer_pair;
 
+    /**
+        Returns the remaining available capacity.
+
+        The returned value represents the available free
+        space in the backing fixed-sized buffers used by the
+        serializer associated with this stream.
+
+        The capacity is absolute and does not do any
+        accounting for any octets required by a chunked
+        transfer encoding.
+    */
     BOOST_HTTP_PROTO_DECL
     std::size_t
     capacity() const noexcept;
 
+    /**
+        Returns the number of octets serialized by this
+        stream.
+
+        The associated serializer stores stream output in its
+        internal buffers. The stream returns the size of this
+        output.
+    */
     BOOST_HTTP_PROTO_DECL
     std::size_t
     size() const noexcept;
 
+    /**
+        Return true if the stream cannot currently hold
+        additional output data.
+
+        The fixed-sized buffers maintained by the associated
+        serializer can be sufficiently full from previous
+        calls to @ref stream::commit.
+
+        This function can be called to determine if the user
+        should drain the serializer via @ref serializer::consume calls
+        before attempting to fill the buffer sequence
+        returned from @ref stream::prepare.
+    */
     BOOST_HTTP_PROTO_DECL
     bool
     is_full() const noexcept;
 
+    /**
+        Returns a MutableBufferSequence for storing
+        serializer input. If `n` bytes are written to the
+        buffer sequence, @ref stream::commit must be called
+        with `n` to update the backing serializer's buffers.
+
+        The returned buffer sequence is as wide as is
+        possible.
+
+        @exception std::length_error Thrown if the stream
+        has insufficient capacity and a chunked transfer
+        encoding is being used
+    */
     BOOST_HTTP_PROTO_DECL
     buffers_type
     prepare() const;
 
+    /**
+        Serialize and commit `n` bytes.
```

Comment:
The term "serialize" is an implementation detail. Me and you know that it is doing the serialization there, but the user is not supposed to know it and in fact it can change later. Maybe we can say:

"Make data available in the serializer's input area."